### ISSUE TITLE
High latency before the first chunk returned in streaming mode

### ISSUE THREAD

**Issue by huskyachao** [FunAudioLLM/CosyVoice#367]
Hi, I found that after updating the code, the latency of the first chunk in streaming mode (inference_zero_shot) is still very high (around 3s-4s). I noticed that issue #294 also mentioned this problem. Is this normal for the current version of cosyvoice? If it is, is there any method to downgrade this latency?

**Comment by aluminumbox** [comment]
yes first chunk inference is slower due to there is no kv cache, also tn takes some time

**Comment by huskyachao** [comment]
> yes first chunk inference is slower due to there is no kv cache, also tn takes some time

OK, I see. Thanks for your answer.

**Comment by MithrilMan** [comment]
@aluminumbox I don't know internal details so this may be a stupid question, but in an use case where the voice is just one (e.g. a local personal assistant), can this be improved by preloading kv cache, or is that cache related to the sentence being spoken and so can't be preloaded and reused every time?
